### ISSUE TITLE
[Bug]: Oracle: datasource not able to connect

### ISSUE THREAD

**Issue by kausstud** [appsmithorg/appsmith#23661]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

``i installed Appsmith on a localhost and tried to connect it with Oracle DB but i got an error. I already connected my local DB with many other apps without and i didn't faced the timezone issue. I only faced it with Appsmith.

ORA-01882 - Timezone is not found (the word is written in Arabic on the picture below)

![timezone issue appsmith](https://github.com/appsmithorg/appsmith/assets/35579004/f3573dad-5933-404b-9272-1084b354a6a8)

### Steps To Reproduce

May you add a button on admin panel to select right timezone for the user so he can make it compatible with the local db.

### Environment

Production

### Version

Self Hosted / Appsmith Community v1.9.19

**Comment by sumitsum** [comment]
Hi @kausstud in case you are using a docker Oracle instance on your local setup, could you please try changing the SSL option to `Disable` ?

**Comment by sumitsum** [comment]
Hi @kausstud thanks for the info, can you please confirm if there is any firewall blocking Appsmith application's request to the Oracle database ?

**Comment by sumitsum** [comment]
@kausstud can you please share which the Oracle database version that you are trying to connect to ?

**Comment by sumitsum** [comment]
Update: this looks related: https://stackoverflow.com/questions/9156379/ora-01882-timezone-region-not-found

**Comment by kausstud** [comment]
Hi @sumitsum 
I checked the firewall and there's nothing blocking Appsmith to connect to the oracle DB.

Version of DB :
Oracle Database 11g Release 11.2.0.1.0 - 64bit Production

i checked the link you have sent for stackoverflow, OK i understand the issue is from the oracle.jdbc.timezoneAsRegion and need to change the value to false.

However , i already connected my oracle DB to many other without any issue , like power bi desktop , retool and i didn't face the timezone issue. 

For me , making changes for the configuration on oracle is really very sensitive step since we have very large records on the DB and i don't want it to be affected by making any new changes on it.

If that is the only way to solve the issue then i won't make it at the moment.

Thank you !

**Comment by sumitsum** [comment]
Thanks for the info @kausstud , can you please share how to get Oracle version 11.2.0.1.0 to test on my local setup ? This way I can do test any fix on my local setup. Otherwise, I can provide a test docker image to you for testing if that's ok.

**Comment by sumitsum** [comment]
@kausstud also, it seems that Oracle 11 is reaching end of support by the end of this year, do you have any plans of migrating to a higher version. I believe this issue should go away once Oracle version is upgraded to the latest i.e. 19+

**Comment by kausstud** [comment]
ok i will tell you what i have faced.

so i have two separate databases, one for factory and another one for store,

the db version for the store is :  Oracle Database 11g Release 11.2.0.1.0 - 64bit Production

the db version for the factory is : Oracle Database 19c Standard Edition 2 Release 19.0.0.0.0 - Production

i just tested the connection for the factory db with Appsmith and worked well with the Oracle Database 19c and you were right the issue is from the old version of oracle 11

but is there any ability to make it work with oracle  11 at the moment ? migrating the DB will be difficult step at this time for me.

I will check with if i found the installation file of oracle 11g version and will send it to  you.

Thank you !

**Comment by sumitsum** [comment]
Hi @kausstud please share the oracle 11g image if possible.

**Comment by sumitsum** [comment]
Closing this issue at the moment, as we may not be able to prioritize support for Oracle version 11g as it has already reached end of life in 2020. In case we can / need to pick this up in future, we can re-open this issue.

**Comment by kausstud** [comment]
Hi @sumitsum 

Sorry for my late reply, i was waiting for the IT department to provide me the installation file of oracle 11g from the sql server.

i have uploaded it on the link below
https://drive.google.com/file/d/1FkrxfXwkyPcONxsMvg0BKGpEgb0gq_6H/view?usp=share_link

hopefully you can find a solution for it asap since i really very interested to link it with our local db.

Thank you for your cooperation.

**Comment by kausstud** [comment]
Hi @sumitsum 

i was able to fix the issue with specifying the Timezone on the running for docker  (  TZ=Asia/Riyadh  ) . i just asked Sage for some help through POE and i really don't know how it worked since i don't have experience with docker. But now i'm able to work on both Factory and store database.  here's the code if you would like to have look .. i run it through git bash

mkdir C:\appsmith-data
docker cp appsmith:/appsmith-data C:\appsmith-data

docker run -it --rm --name mycontainer -v /c/Users/user/appsmith-data:/appsmith-data -p 8080:80 -e TZ=Asia/Riyadh appsmith/appsmith-ce


I only faced this issue with Appsmith. I worked with many other lowcode and was able to connect both databases successfully without an issue.

Hope this helps you to fix it on the original files. 

Thank you !
